src/Filters/stream_filt.jl

```diff
@@ -552,6 +553,9 @@ function filt!(buffer::AbstractVector{Tb}, self::FIRFilter{FIRDecimator{Th}}, x:
     outLen              = outputlength(self, xLen)
     inputIdx            = kernel.inputDeficit
 
+    nbufout = fld(xLen - inputIdx, kernel.decimation)
```

Comment:
If `xLen == inputIdx`, one value is stored to `buffer` below, so this can't be quite right. Might be just missing a `+ 1`, but I haven't really thought this through.